src/client.rs

```diff
@@ -55,70 +58,54 @@ impl Client {
     /// Returns an async connection from the client.
     #[cfg(feature = "aio")]
     pub async fn get_async_connection(&self) -> RedisResult<crate::aio::Connection> {
-        #[cfg(all(feature = "tokio-comp", not(feature = "async-std-comp")))]
-        {
-            self.get_tokio_connection_tokio().await
-        }
-
-        #[cfg(all(not(feature = "tokio-comp"), feature = "async-std-comp"))]
-        {
-            self.get_async_std_connection().await
-        }
-
-        #[cfg(all(feature = "tokio-comp", feature = "async-std-comp"))]
-        {
-            if tokio::runtime::Handle::try_current().is_ok() {
-                self.get_tokio_connection_tokio().await
-            } else {
-                self.get_async_std_connection().await
+        let con = match self.get_runtime() {
+            #[cfg(feature = "tokio-comp")]
+            Runtime::Tokio => {
+                self.get_simple_async_connection::<crate::aio::tokio_aio::Tokio>()
+                    .await?
             }
-        }
+            #[cfg(feature = "async-std-comp")]
+            Runtime::AsyncStd => {
+                self.get_simple_async_connection::<crate::aio_async_std::AsyncStd>()
+                    .await?
+            }
+        };
 
-        #[cfg(all(not(feature = "tokio-comp"), not(feature = "async-std-comp")))]
-        {
-            compile_error!("tokio-comp or async-std-comp features required for aio feature")
-        }
+        crate::aio::Connection::new(&self.connection_info, con).await
     }
 
     /// Returns an async connection from the client.
     #[cfg(feature = "tokio-comp")]
-    pub async fn get_tokio_connection_tokio(&self) -> RedisResult<crate::aio::Connection> {
-        crate::aio::connect_tokio(&self.connection_info).await
+    pub async fn get_tokio_connection(&self) -> RedisResult<crate::aio::Connection> {
+        use crate::aio::RedisRuntime;
+        Ok(
+            crate::aio::connect::<crate::aio::tokio_aio::Tokio>(&self.connection_info)
+                .await?
+                .map(RedisRuntime::boxed),
+        )
     }
 
     /// Returns an async connection from the client.
     #[cfg(feature = "async-std-comp")]
     pub async fn get_async_std_connection(&self) -> RedisResult<crate::aio::Connection> {
-        crate::aio::connect_async_std(&self.connection_info).await
+        use crate::aio::RedisRuntime;
+        Ok(
+            crate::aio::connect::<crate::aio_async_std::AsyncStd>(&self.connection_info)
+                .await?
+                .map(RedisRuntime::boxed),
+        )
     }
 
     /// Returns an async connection from the client.
     #[cfg(all(feature = "aio", feature = "tokio-rt-core"))]
```

Comment:
Should either require `tokio-rt-core` or `async-std-comp` but not always `tokio-rt-core`.

```suggestion
    #[cfg(any(feature = "tokio-rt-core", all(not(feature = "tokio-comp"), feature = "async-std-comp")))]
```